zio-http/src/test/scala/zio/http/ClientProxySpec.scala

```diff
@@ -49,6 +49,44 @@ object ClientProxySpec extends HttpRunnableSpec {
         } yield out
       assertZIO(res.either)(isLeft(isSubtype[ConnectException](anything)))
     },
+    test("ZClient proxy respond Ok") {
+
+      def proxied(proxy: Proxy): ZLayer[Config with ClientDriver with DnsResolver, Throwable, Client] = {
+        implicit val trace: Trace = Trace.empty
+        ZLayer.scoped {
+          for {
+            config <- ZIO.service[Config]
+            driver <- ZIO.service[ClientDriver]
+            dnsResolver <- ZIO.service[DnsResolver]
+            connectionPool <- driver.createConnectionPool(dnsResolver, config.connectionPool)
+            baseClient = ZClient.fromDriver(new DriverLive(driver)(connectionPool)(config))
+          } yield
+            baseClient.copy(proxy = Some(proxy))
+        }
+      }
+
+      val res =
+        for {
+          port <- ZIO.environmentWithZIO[DynamicServer](_.get.port)
+          url <- ZIO.fromEither(URL.decode(s"http://localhost:$port"))
+          id <- DynamicServer.deploy(Handler.ok.toHttpApp)
+          proxy = Proxy.empty.url(url).headers(Headers(DynamicServer.APP_ID, id))
+
+          out <- Client
+            .request(
+              Request.get(url = url),
+            )
+            .provide(
+              ZLayer.succeed(ZClient.Config.default),
+              proxied(proxy),
```

Comment:
@adamgfraser I don't know how to reuse `ZClient.customized` or using `NettyClientDriver.live` to create proxy ZClient. With this I ended up with 2 `ClientDriver`s. 

Need your feedback on refactor/improve this unit test

Reply:
@rajcspsg I think it would be helpful to take a look at the text below `proxy respond Ok` and think about the user feature we are trying to implement and how the test could be rewritten to use our functionality.

In that test, the client is created with `Client.customized`, which requires a `Config`, and the `Config` is customized to use a proxy by using the `proxy` operator on `Config`.

The functionality we are trying to implement is to allow the user to customize the client locally. So instead of having to provide the proxy at the very top of the application the user should be able to specify it right before they call an operator on the client or at any other level.

This means that instead of providing `Client.customized` the user should be able to provide `Client.default` or leave the `Config` as `Config.default`.

Then in your test you can call `ZIO.service[Client]` to access the client, call your new `proxy` operator on `Client` to get a customized client that uses the proxy and then get the same result as in the original test.

Reply:
@adamgfraser I updated the code. Could you please take a look again?

Reply:
@adamgfraser could you please take at my code changes?